src/components/RichEditor/index.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Editor } from '@tinymce/tinymce-react';
+import 'tinymce/tinymce.min';
+import 'tinymce/plugins/legacyoutput';
+import 'tinymce/plugins/link';
+import 'tinymce/plugins/lists';
+import 'tinymce/themes/silver/theme';
+import 'style-loader!tinymce/skins/ui/oxide/skin.min.css'; // eslint-disable-line import/no-webpack-loader-syntax, import/no-unresolved
+
+
+class RichEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      charCount: 0,
+    };
+    this.initCharCount = this.initCharCount.bind(this);
+    this.updateCharCount = this.updateCharCount.bind(this);
+  }
+
+  initCharCount(event, editor) {
+    const content = editor.getContent({ format: 'text' });
+    this.setState({
+      charCount: content.length,
+    });
+  }
+
+  updateCharCount(event, editor) {
+    const content = editor.getContent({ format: 'text' });
+    this.setState({
+      charCount: content.length,
+    });
+    const htmlContent = editor.getContent();
+    this.props.input.onChange(htmlContent);
+  }
+
+  render() {
+    const remainingChars = this.props.maxChars - this.state.charCount;
+    const characterLimitMessage = `Recommended character limit (including spaces) is
+      ${this.props.maxChars}. ${remainingChars} characters remaining.`;
```

Comment:
```this.props.maxChars``` is used more than once in the render function so I would recommend destructuring it.